### PR TITLE
orion in ha on docker swarm

### PR DESCRIPTION
**Pull request by chicco785**
this pull request add to the documentation a short introduction on deploying orion in ha on top of a docker swarm cluster

### CODE REVIEW

**Comment by chicco785** [comment]
@jmcanterafonseca as promised ;)

File: docker/DOCKERSWARM.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+# How to deploy Orion Context Broker in a Docker Swarm cluster
```
Comment (fgalan) [inline docker/DOCKERSWARM.md]: File name: DOCKERSWARM.md -> dockerSwarm.md or docker_swarm.md

File: README.md [hunk 2]
```diff
@@ -153,6 +153,7 @@ Please have a look to the section [on building the source code](doc/manuals/admi
 * [Installation and administration](doc/manuals/admin/README.md)
 * Container-based deployment
   * [Docker](docker/README.md)
+  * [Docker Swarm and HA](docker/README.md)
```
Comment (fgalan) [inline README.md]: Broken link

File: docker/DOCKERSWARM.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Here we provide only a quick overview.
+
+## Installing a Docker Swarm test cluster
+
+To deploy on your linux based system a Docker Swarm test cluster, you can used
+[miniswarm](https://github.com/aelsabbahy/miniswarm).
+
+The only pre-requisite are:
+* [Docker Machine](https://docs.docker.com/machine/install-machine/)
+* [Virtual Box](http://virtualbox.org/).
```
Comment (fgalan) [inline docker/DOCKERSWARM.md]: Remove . at the end of the line

**Comment by fgalan** [comment]
Some of the heading in markdown seems are not working. Looking to https://github.com/martel-innovate/fiware-orion/blob/docker-swarm-doc/docker/DOCKERSWARM.md I see the following ones:

```
### Connect to the cluster
### Deploy a MongoDB ReplicaSet cluster
### Deploy a Orion Context Broker cluster
```

In addition, all the numbered bullets are using the same number (1)

File: docker/DOCKERSWARM.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+   that offers additional functionalities such as healthchecks and use separate
+   networks for frontend and backend):
+
+  ```yaml
+  version: '3'
+
+  services:
+
+    orion:
+      image: fiware/orion:1.7.0
```
Comment (fgalan) [inline docker/DOCKERSWARM.md]: This piece of documentation should survive to the pass of the year ;). Thus, I'll recommend using:

```
      image: fiware/orion:latest
```

**Comment by fgalan** [comment]
Nice pieces of documentation... Thanks @chicco785 !

I have provides some minor comments on it. Let's wayt also for @jmcanterafonseca review, as I understand he is involved in this in some way or another.

**Comment by fgalan** [comment]
LGTM. Thanks!

Passing the ball to @jmcanterafonseca for additional LGTM

File: docker/docker_swarm.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+cluster management solution and it is natively integrated with Docker Engine.
+
+You can find detailed instructions and consideration on running Orion Context
+Broker on Docker Swarm at this [link](https://smartsdk.github.io/smartsdk-recipes/data-management/context-broker/ha/readme/).
+
+Here we provide only a quick overview.
+
+## Installing a Docker Swarm test cluster
+
+To deploy on your linux based system a Docker Swarm test cluster, you can used
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: Linux

Comment (jmcanterafonseca) [reply docker/docker_swarm.md]: you can use

File: docker/docker_swarm.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+```bash
+# 1 manager 2 workers
+$ miniswarm start 3
+```
+
+### Connect to the cluster
+
+```bash
+$ eval $(docker-machine env ms-manager0)
+```
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: what should be the result of this instruction? Please could you put an example?

File: docker/docker_swarm.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+## Deploy Orion Context Broker in HA
+
+To deploy Orion in High Availability, you need first to deploy a MongoDB
+ReplicaSet, and then deploy on top of it your Orion Context Broker.
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: please could you add a link on how to deploy a MongoDB ReplicaSet ?

File: docker/docker_swarm.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+          - mongodata:/data/db
+        networks:
+          - backend
+        deploy:
+          mode: global
+          restart_policy:
+            condition: on-failure
+          update_config:
+            parallelism: 1
+            delay: 1m30s
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: why this delay is needed?

File: docker/docker_swarm.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+      controller:
+        image: martel/mongo-replica-ctrl:latest
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: what's the purpose of this image?

File: docker/docker_swarm.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+    ```bash
+    $ docker stack deploy -c docker-compose-orion.yml orion
+    ```
+
+1. Check that the deployment worked out correctly:
+
+    ```bash
+    $ curl $(docker-machine ip ms-manager0):1026/v2/entities -s -S --header 'Accept: application/json' --header 'fiware-service: default' --header 'fiware-ServicePath: /'
+    ```
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: please show the result of this command

File: docker/docker_swarm.md [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+
+1. Scale up orion:
+
+    ```bash
+    $ docker service scale orion_orion=3
+    ```
```
Comment (jmcanterafonseca) [inline docker/docker_swarm.md]: please show the result of this command

**Comment by jmcanterafonseca** [comment]
general comment: I think there should be a link between this piece of documentation and the one detailing the HA architecture with the HA Proxy.

**Comment by chicco785** [comment]
also this one should be ready for landing.

**Pull request merged**